### DISCUSSION TITLE
Is there need to cleanup eventListeners added by d3?

### DISCUSSION THREAD

**Discussion by neznayer** [d3/d3#3739] [Q&A]
I'm using d3.js with Svelte, inside a custom svelte action. I'm adding some mouse event listeners with d3, like 

```js
rect.on("mouseenter", function(event, d) {...} )
```

The svelte action optionally can return a cleanup function which is executed when the node is unmounted from DOM. 

Is there a need to remove eventListeners added with d3 with `.on(...)` on node unmount? If so, how to do that?

**Comment by adebiyial** [comment]
@neznayer You should. See [onDestroy](https://svelte.dev/docs/svelte#ondestroy)

```
onDestroy(() => {
  // Remove event listener when the component is unmounted
  rect.on('mouseenter', null);
});
```

**Comment by curran** [comment]
What happens if you don't remove the listener?

**Reply by christopherreay** [reply]
so the JavaScript never reloads, right, the page runtime,

so
.... actually, i don't know.

when items are removed from the DOM i think event listeners are removed. so the question is, is there some other reference to the function somewhere in the scope?

i do not know the answer to this
